docs/core/diagnostics/debug-highcpu.md

```diff
@@ -83,32 +83,50 @@ With the web app running, immediately after startup, the CPU isn't being consume
 
 ```dotnetcli
-dotnet-counters monitor --counters System.Runtime[cpu-usage] -p 22884 --refresh-interval 1
+dotnet-counters monitor -p 22884 --refresh-interval 1
 ```
 
-You should see an increase in CPU usage as shown below:
+You should see an increase in CPU usage as shown below (depending on the host machine, expect varying CPU usage):
 
 ```console
 Press p to pause, r to resume, q to quit.
     Status: Running
 
 [System.Runtime]
+    % Time in GC since last GC (%)                         0
+    Allocation Rate / 1 sec (B)                            0
     CPU Usage (%)                                         25
+    Exception Count / 1 sec                                0
+    GC Heap Size (MB)                                      4
+    Gen 0 GC Count / 60 sec                                0
+    Gen 0 Size (B)                                         0
+    Gen 1 GC Count / 60 sec                                0
+    Gen 1 Size (B)                                         0
+    Gen 2 GC Count / 60 sec                                0
+    Gen 2 Size (B)                                         0
+    LOH Size (B)                                           0
+    Monitor Lock Contention Count / 1 sec                  0
+    Number of Active Timers                                1
+    Number of Assemblies Loaded                          140
+    ThreadPool Completed Work Item Count / 1 sec           3
+    ThreadPool Queue Length                                0
+    ThreadPool Thread Count                                7
+    Working Set (MB)                                      63
 ```
 
-Throughout the duration of the request, the CPU usage will hover around 25% . Depending on the host machine, expect varying CPU usage.
+Throughout the duration of the request, the CPU usage will hover around the increased percentage.
 
 > [!TIP]
 > To visualize an even higher CPU usage, you can exercise this endpoint in multiple browser tabs simultaneously.
 
-At this point, you can safely say the CPU is running higher than you expect.
+At this point, you can safely say the CPU is running higher than you expect. Identifying the effects of a problem is key to finding the cause. We will use the effect of high CPU consumption in addition to diagnostic tools to find the cause of the problem.
 
-## Trace generation
+## Analyze High CPU with Profiler
 
-When analyzing a slow request, you need a diagnostics tool that can provide insights into what the code is doing. The usual choice is a profiler, and there are different profiler options to choose from.
+When analyzing an app with high CPU usage, you need a diagnostics tool that can provide insights into what the code is doing. The usual choice is a profiler, and there are different profiler options to choose from.
 
 ### [Linux](#tab/linux)
 
```

Comment:
Should we explain why we suggest using the Linux perf tool and not dotnet-trace?  I'm guessing the reason is something like: Although Dotnet-Trace is cross-platform but since Event Pipe "CPU" traces only measure elapsed time and not actual CPU time.

Reply:
I had the same question. @josalem Why are we using the Linux perf tool and not dotnet-trace?

Reply:
`dotnet-trace` is great for CPU investigations to a certain point. The limitations of safe-point bias and managed-only callstacks means that it can't give _exact_ information the same way a kernel-aware profiler like `perf` can.

If your performance investigation stops at managed code, generally `dotnet-trace` will be sufficient. If you are analyzing exact timing of your application and it involves native code, you may need something like `perf`. 

I think of it like two differently powered magnifying glasses. You can start with `dotnet-trace` and then look at a higher resolution using `perf` (or `ETW` on Windows).